webpack.config.js

```diff
@@ -0,0 +1,118 @@
+const webpack = require("webpack");
+const path = require("path");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const MonacoWebpackPlugin = require("monaco-editor-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+const ServiceWorkerWebpackPlugin = require("serviceworker-webpack-plugin");
+const TerserPlugin = require('terser-webpack-plugin');
+
+const config = {
+  entry: {
+    main: ["babel-polyfill", "./js/MainPage/index.js"]
+  },
+  output: {
+    path: path.resolve(__dirname, "dist"),
+    filename: "[name].bundle[hash].js"
+  },
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        use: "babel-loader",
+        exclude: /node_modules/
+      },
+      {
+        test: /\.css$/,
+        use: [
+          "style-loader",
+          {
+            loader: "css-loader",
+            options: {
+              importLoaders: 1
+            }
+          },
+          "postcss-loader"
+        ]
+      },
+      {
+        test: /\.(woff(2)?|ttf|eot)(\?v=\d+\.\d+\.\d+)?$/,
+        use: [
+          {
+            loader: "file-loader",
+            options: {
+              name: "[name].[ext]",
+              outputPath: "fonts/"
+            }
+          }
+        ]
+      },
+      {
+        test: /opencascade\.wasm\.wasm$/,
+        type: "javascript/auto",
+        loader: "file-loader"
+      }
+    ]
+  },
+  node: {
+    fs: "empty"
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      template: path.resolve(__dirname, "index.html")
```

Comment:
Because webpack primarily works for js files, this plugin is basically here to tell it to put index.html into the dist folder (the built folder that will be exposed to the web) and it adds a script tag to load our entry `index.js` file.

Reply:
I can see the confused emoji @zalo, I'll try to clarify but if I miss the mark let me know what you confused by and  I might be able to provide better context.

If you build the production files by `yarn build-prod` it will put a bunch of files into `/dist` (and that's what vercel will serve up for us).

There are a bunch files in there, so `runtime.bundle[hash].js` is the main bundle (hashes are added for cache invalidation reasons I believe), the hash makes it difficult for us to include the bundle manually though, so letting webpack handle putting index.html into `dist` means it can include this file, so if you look at `index.html` in `/dist` folder, you see a script tag for the runtime bundle.